### DISCUSSION TITLE
Download BRouter segments

### DISCUSSION THREAD

**Discussion by Leeb64** [devemux86/cruiser#443] [Q&A] [answered]
If after downloading individual Brouter segments the route calculation with the BROUTER no longer works correctly, ALL segments must be downloaded again.
It just happened to me again.

**Comment by devemux86** [comment]
@Leeb64 thanks for the report.

Internal or external BRouter? Is there any message?

If it happens again, check the content of "/brouter/modes/" folder (inside the selected "Storage folder").
See if there are any *.dat files and with what size, especially if anyone has 0 size.

In the current version, just re-download 1 segment and it will work again.

---

There is a bug in BRouter which I have already sent the fix and it will also be included in the next Cruiser (after 40):
- https://github.com/abrensch/brouter/issues/671

cc @afischerdev

**Comment by devemux86** [comment]
Another case is if the downloads stopped somehow?
So the downloaded files were not correct and could not be used in routing.

**Reply by Leeb64** [reply]
With the internal brouter.
It's happened to me for the second time.
-Just in case anyone has the same problems!

**Reply by devemux86** [reply]
@Leeb64 please watch carefully what you did and report the steps if possible,
so we can understand what is happening.

**Reply by Leeb64** [reply]
Although it works for short routes,
On longer routes, from around 200km onwards, the Brouter stops after around 30s.

**Reply by devemux86** [reply]
@Leeb64 

What does route length have to do with the downloaded data?
There is no relation between them at all. You probably mean something else?

If a route cannot be calculated, please post the route and the routing options.
(so we can check it)

**Reply by Leeb64** [reply]
No idea.
It still works for route lengths of around 150-200km. Then comes the message that the calculation is not possible and intermediate points should be added.
After downloading the complete Brouter segments again, everything works perfectly again.

**Comment by devemux86** [comment]
For very long routes, it is recommended to **add many waypoints**.
Then the calculation speed will be greatly improved.

---

Also we will have soon new route calculation modes in Cruiser:
- https://github.com/devemux86/cruiser/discussions/142

- https://github.com/devemux86/cruiser/discussions/364

**Reply by Leeb64** [reply]
That's already clear.
Nevertheless, something seems to go wrong every now and then.
As I said: after renewing all segments it works.

**Comment by devemux86** [comment]
@Leeb64 please try with the new [Cruiser 3.0.40](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

It contains the fix for another BRouter report (may be a similar case):

- https://github.com/devemux86/cruiser/discussions/438

**Comment by devemux86** [comment]
@Leeb64 please try:
- https://github.com/devemux86/cruiser/discussions/451

**Comment by devemux86** [comment, marked as answer]
Fixed in [Cruiser 3.0.41](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
